src/Neo/Network/P2P/Payloads/Signer.cs

```diff
@@ -72,18 +72,26 @@ public bool Equals(Signer other)
         {
             if (ReferenceEquals(this, other))
                 return true;
+
             if (other is null) return false;
-            return Account == other.Account &&
-                Scopes == other.Scopes &&
-                AllowedContracts.AsSpan().SequenceEqual(other.AllowedContracts.AsSpan()) &&
-                AllowedGroups.AsSpan().SequenceEqual(other.AllowedGroups.AsSpan()) &&
-                Rules.AsEnumerable().SequenceEqual(other.Rules.AsEnumerable());
+            if (Account != other.Account || Scopes != other.Scopes)
+                return false;
+
+            if (Scopes.HasFlag(WitnessScope.CustomContracts) && !AllowedContracts.SequenceEqual(other.AllowedContracts))
```

Comment:
problem was `AllowedContracts`, `AllowedGroups`, and `Rules` being  `null` 

Example `this == null`

Reply:
Same logic as serialize and deserialize, If it's well constructed, it will work